packages/trimerge-sync/src/lib/PrefixLogger.ts

```diff
@@ -0,0 +1,36 @@
+import { Logger } from '../types';
+
+/** Utility that applies a prefix to log messages. If provided with a Prefix logger, it will concatenate the prefixes. */
+export class PrefixLogger implements Logger {
+  readonly prefix: string;
+  readonly logger: Logger;
+  constructor(prefix: string, logger: Logger) {
+    if (logger instanceof PrefixLogger) {
+      this.prefix = `${logger.prefix}/${prefix}`;
+      this.logger = logger.logger;
+    } else {
+      this.prefix = prefix;
+      this.logger = logger;
+    }
+  }
+
+  debug(...args: any[]) {
```

Comment:
there's a syntax in console.log that lets you do formatted strings, so you might want to do:
```suggestion
  debug(first: unknown, ...rest: unknown[]) {
```

Reply:
I like that idea! I'll leave that for a future PR though.